continuousflex/protocols/protocol_pdb_dimred.py

```diff
@@ -74,111 +61,101 @@ def _defineParams(self, form):
                       condition='pdbSource == 1',
                       label="List of PDBs",
                       help='Use the file pattern as file location with /*.pdb')
-        form.addParam('dimredMethod', EnumParam, default=DIMRED_SKLEAN_PCA,
-                      choices=['Principal Component Analysis (PCA)',
-                               'Local Tangent Space Alignment',
-                               'Diffusion map',
-                               'Linear Local Tangent Space Alignment',
-                               'Linearity Preserving Projection',
-                               'Kernel PCA',
-                               'Probabilistic PCA',
-                               'Laplacian Eigenmap',
-                               'Hessian Locally Linear Embedding',
-                               'Stochastic Proximity Embedding',
-                               'Neighborhood Preserving Embedding',
-                               'Scikit-Learn PCA',
-                               "Don't reduce dimensions"],
-                      label='Dimensionality reduction method',
-                      help=""" Choose among the following dimensionality reduction methods:
-            PCA
-               Principal Component Analysis 
-            LTSA <k=12>
-               Local Tangent Space Alignment, k=number of nearest neighbours 
-            DM <s=1> <t=1>
-               Diffusion map, t=Markov random walk, s=kernel sigma 
-            LLTSA <k=12>
-               Linear Local Tangent Space Alignment, k=number of nearest neighbours 
-            LPP <k=12> <s=1>
-               Linearity Preserving Projection, k=number of nearest neighbours, s=kernel sigma 
-            kPCA <s=1>
-               Kernel PCA, s=kernel sigma 
-            pPCA <n=200>
-               Probabilistic PCA, n=number of iterations 
-            LE <k=7> <s=1>
-               Laplacian Eigenmap, k=number of nearest neighbours, s=kernel sigma 
-            HLLE <k=12>
-               Hessian Locally Linear Embedding, k=number of nearest neighbours 
-            SPE <k=12> <global=1>
-               Stochastic Proximity Embedding, k=number of nearest neighbours, global embedding or not 
-            NPE <k=12>
-               Neighborhood Preserving Embedding, k=number of nearest neighbours 
-        """)
-        form.addParam('extraParams', params.StringParam, default=None,
-                      expertLevel=params.LEVEL_ADVANCED,
-                      label='Extra params',
-                      help='These parameters are there to change the default parameters of a dimensionality reduction'
-                           ' method. Check xmipp_matrix_dimred for full details.')
-
+        form.addParam('setOfPDBs', params.PointerParam, pointerClass='SetOfPDBs, SetOfAtomStructs',
+                      condition='pdbSource == 2',
```

Comment:
Since you have already defined global variables for different pdbSource, you can use it in the conditions
like `condition= "pdbSource == %d" %PDB_SOURCE_SUBTOMO`

Reply:
Done